JoonasHeinonen/escape-from-zordoom-prison
# Feature/explosive crate

### [Explosive crate](https://trello.com/c/VQKyjRiH/52-explosive-crate)


https://user-images.githubusercontent.com/33086486/210118671-5c4de98d-7936-41bf-b41c-4fc3f911f5e9.mp4

**Changes:**
 - Imported the model for the explosive crate and the materials needed.
 - Added animations to the explosive crate. Once activated, it will count to 3 with an animation for visual help. Once at its end, an explosion erupts.
 - Added a particle scene for the explosion, with lighting effect.

**How to test:**
 - You may add any number of explosive crates in order to see the functionality in its full extent.
 - Shoot any explosive crate, next to nothing or to other crates, bolt crates and / or other explosive crates, for instance.
 - See that the explosion radius either destroys the other bolt crates or explosive crates.
 - See that it's working to its best extent.

**Sources cited:**
 - [Timer](https://docs.godotengine.org/en/stable/classes/class_timer.html)
 - [Particle Systems From BEGINNER To PRO | Godot Engine](https://www.youtube.com/watch?v=F1Fyj3Lh_Pc&t=239s&ab_channel=PlayWithFurcifer)
 - [How to make a simple fire in Godot with 3D particles](https://www.youtube.com/watch?v=8ngg6ueC_ks&t=162s&ab_channel=Miziziziz)
